L1Trigger/TrackFindingTMTT/src/TP.cc

```diff
@@ -0,0 +1,189 @@
+      }
+    }
+
+    this->fillUseForAlgEff();  // Fill useForAlgEff_ flag.
+
+    this->calcNumLayers();  // Calculate number of tracker layers this TP has stubs in.
+  }
+
+  //=== Check if this tracking particle is worth keeping.
+  //=== (i.e. If there is the slightest chance of reconstructing it, so as to measure fake rate).
+
+  void TP::fillUse() {
+    const bool useOnlyInTimeParticles = false;
+    const bool useOnlyTPfromPhysicsCollisionFalse = false;
+    // Use looser cuts here those those used for tracking efficiency measurement.
+    // Keep only those TP that have a chance (allowing for finite track resolution) of being reconstructed as L1 tracks. L1 tracks not matching these TP will be defined as fake.
+
+    const vector<int> genPdgIdsAllUnsigned = {
+        11,
+        13,
+        211,
+        321,
+        2212};  // Include all possible particle types here, as if some are left out, L1 tracks matching one of missing types will be declared fake.
+    vector<int> genPdgIdsAll;
+    for (unsigned int i = 0; i < genPdgIdsAllUnsigned.size(); i++) {
+      genPdgIdsAll.push_back(genPdgIdsAllUnsigned[i]);
+      genPdgIdsAll.push_back(-genPdgIdsAllUnsigned[i]);
+    }
+
+    // Range big enough to include all TP needed to measure tracking efficiency
+    // and big enough to include any TP that might be reconstructed for fake rate measurement.
+    const float ptMin = min(settings_->genMinPt(), 0.7 * settings_->houghMinPt());
+    const float etaMax = max(settings_->genMaxAbsEta(), 0.2 + fabs(settings_->etaRegions()[0]));
+
+    static TrackingParticleSelector trackingParticleSelector(ptMin,
+                                                             9999999999,
+                                                             -etaMax,
+                                                             etaMax,
+                                                             max(10.0, settings_->genMaxVertR()),
+                                                             max(35.0, settings_->genMaxVertZ()),
+                                                             0,
+                                                             useOnlyTPfromPhysicsCollisionFalse,
+                                                             useOnlyInTimeParticles,
+                                                             true,
+                                                             false,
+                                                             genPdgIdsAll);
+
+    const TrackingParticlePtr tp_ptr(*this);  // cast to base class.
+    use_ = trackingParticleSelector(*tp_ptr);
+  }
+
+  //=== Check if this tracking particle can be used to measure the L1 tracking efficiency.
+
+  void TP::fillUseForEff() {
+    useForEff_ = false;
+    if (use_) {
+      const bool useOnlyInTimeParticles = true;
+      const bool useOnlyTPfromPhysicsCollision = true;
+      static TrackingParticleSelector trackingParticleSelector(settings_->genMinPt(),
```

Comment:
```suggestion
      static const TrackingParticleSelector trackingParticleSelector(settings_->genMinPt(),
```
although I believe some other solution for the `TrackingParticleSelector` would work better in the long run.

Reply:
done